src/engine/Collision/CollisionGroupManager.ts

```diff
@@ -0,0 +1,52 @@
+import { CollisionGroup } from './CollisionGroup';
+
+/**
+ * Static class for managing collision groups in excalibur, there is a maximum of 32 collision groups possible in excalibur
+ */
+export class CollisionGroupManager {
+  // using bitmasking the maximum number of groups is 32, because that is the heighest 32bit integer that JS can present.
+  private static _MAX_GROUPS = 32;
+  private static _currentGroup = 1;
+  private static _currentBit = 0b1 | 0;
+  private static _groups: { [name: string]: CollisionGroup } = {};
```

Comment:
I propose using native `Map` as it's waaaaaaaay faster: https://jsperf.com/map-values-vs-object-values/1

Reply:
In fact when/if you change it, you should measure what the impact is 👍

Reply:
I'll make the change later today